### ISSUE TITLE
New Rule: Forbidden the case that parent class define a setter and inherited class define a getter (vice versa)

### ISSUE THREAD

**Issue by yf-yang** [eslint/eslint#17079]
### Rule details

When an inherited class declare a setter only, check if one of its ancestors declare a getter. When an inherited class declare a getter only, check if one of its ancestors declare a setter. If so, complains.

### Related ECMAScript feature

check https://stackoverflow.com/a/60936882/5930115, it provides references

### What type of rule is this?

Warns about a potential problem

### Example code

```js
class ParentClass {
    get property(value) {
        return this.property_;
    }
}

class Subclass extends ParentClass {
    set property(value) {
        this.property_ = value;
    }
}
```

### Why should this rule be in the core instead of a plugin?

It's a common feature of ecmascript class.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

I encountered this bug and have no idea what happens, then found those two stack overflow questions:
https://stackoverflow.com/questions/60899900/why-the-getters-or-the-setters-are-not-inherited-independently-in-javascript
https://stackoverflow.com/questions/28950760/override-a-setter-and-the-getter-must-also-be-overridden

I believe most of the people would assume all methods of a class are independently inherited, even if two methods are getters and setters of the same property. However, as the spec shows, getter and setter are actually coupled, and people hardly know it.
